### PR TITLE
Clarify and fix docs about supported WebSocket subprotocols

### PR DESCRIPTION
**Pull request by shorgi**
From previous discussion, fix docs about:
- the default subprotocol
- requiring absolute paths for protocol config 

Deploy preview: 
- Updated [Websocket setup](https://deploy-preview-4063--apollo-router-docs.netlify.app/executing-operations/subscription-support#websocket-setup)

### CODE REVIEW

File: docs/source/executing-operations/subscription-support.mdx [hunk 1]
```diff
@@ -146,30 +146,34 @@ subscription:
...
-  - Legacy subprotocol used by the [`subscriptions-transport-ws` library](https://github.com/apollographql/subscriptions-transport-ws), which is **unmaintained**
-  - For backward compatibility, this is the default value.
 - `graphql_transport_ws`
   - Used by the [graphql-ws](https://github.com/enisdenjo/graphql-ws) library
-  - This subprotocol is recommended for GraphQL server libraries implementing WebSocket-based subscriptions.
-
-> ⚠️ **Your router must use whichever subprotocol is expected by each of your subgraphs!**
+  - This subprotocol is the **default value** and is recommended for GraphQL server libraries implementing WebSocket-based subscriptions.
+- `graphql_ws`
+  - Legacy subprotocol used by the [`subscriptions-transport-ws` library](https://github.com/apollographql/subscriptions-transport-ws), which is **unmaintained** but provided for backward compatibility.
```
Comment (lleadbet) [inline docs/source/executing-operations/subscription-support.mdx:164]: We need to flip the bullets here- `graphql_ws` is for `graphql-ws` library, and vice versa.

File: docs/source/executing-operations/subscription-support.mdx [hunk 2]
```diff
@@ -146,30 +146,34 @@ subscription:
...
 - `graphql_transport_ws`
   - Used by the [graphql-ws](https://github.com/enisdenjo/graphql-ws) library
-  - This subprotocol is recommended for GraphQL server libraries implementing WebSocket-based subscriptions.
-
-> ⚠️ **Your router must use whichever subprotocol is expected by each of your subgraphs!**
+  - This subprotocol is the **default value** and is recommended for GraphQL server libraries implementing WebSocket-based subscriptions.
+- `graphql_ws`
+  - Legacy subprotocol used by the [`subscriptions-transport-ws` library](https://github.com/apollographql/subscriptions-transport-ws), which is **unmaintained** but provided for backward compatibility. 
 
-By default, the router uses `graphql_ws` for all subgraphs. You can change this global default and/or override it for individual subgraphs by setting the `protocol` key as shown above.
```
Comment (lleadbet) [inline docs/source/executing-operations/subscription-support.mdx:169]: This is actually the current (and preferred) behavior.

**Review by lleadbet — CHANGES_REQUESTED** [review]

File: docs/source/executing-operations/subscription-support.mdx [hunk 3]
```diff
@@ -146,30 +146,34 @@ subscription:
...
+By default, the router uses the `graphql_transport_ws` protocol option for all subgraphs. You can change this global default and/or override it for individual subgraphs by setting the `protocol` key as shown above.
 
 Your router creates a separate WebSocket connection for each client subscription, _unless_ it can perform [subscription deduplication](#subscription-deduplication).
 
+> ⚠️ **Notes:**
+> * Your router must use whichever subprotocol is expected by each of your subgraphs.
+> * To disambiguate between `graph-ws` and `graph_ws`: 
+>    * `graph-ws` (with a dash) is the name of the [library](https://github.com/enisdenjo/graphql-ws) that uses the recommended `graphql_transport_ws` WebSocket subprotocol.
```
Comment (Meschreiber) [inline docs/source/executing-operations/subscription-support.mdx:173]: I always find it nice to include example characters to be clear

```suggestion
>    * `graph-ws` (with a hyphen `-`) is the name of the [library](https://github.com/enisdenjo/graphql-ws) that uses the recommended `graphql_transport_ws` WebSocket subprotocol. 
```

File: docs/source/executing-operations/subscription-support.mdx [hunk 4]
```diff
@@ -146,30 +146,34 @@ subscription:
...
-By default, the router uses `graphql_ws` for all subgraphs. You can change this global default and/or override it for individual subgraphs by setting the `protocol` key as shown above.
+By default, the router uses the `graphql_transport_ws` protocol option for all subgraphs. You can change this global default and/or override it for individual subgraphs by setting the `protocol` key as shown above.
 
 Your router creates a separate WebSocket connection for each client subscription, _unless_ it can perform [subscription deduplication](#subscription-deduplication).
 
+> ⚠️ **Notes:**
+> * Your router must use whichever subprotocol is expected by each of your subgraphs.
+> * To disambiguate between `graph-ws` and `graph_ws`: 
+>    * `graph-ws` (with a dash) is the name of the [library](https://github.com/enisdenjo/graphql-ws) that uses the recommended `graphql_transport_ws` WebSocket subprotocol. 
+>     * `graph_ws` (with an underscore) is the router's configuration value to use the legacy subprotocol of the `subscriptions-transport-ws` library.
```
Comment (Meschreiber) [inline docs/source/executing-operations/subscription-support.mdx:174]:
```suggestion
>     * `graph_ws` (with an underscore `_`) is the router's configuration value to use the legacy subprotocol of the `subscriptions-transport-ws` library.
```

File: docs/source/executing-operations/subscription-support.mdx (hunk 1 above)
Comment (shorgi) [reply docs/source/executing-operations/subscription-support.mdx:164]: ugh, I got tripped up after reading again https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#communication stating `The WebSocket sub-protocol for this specification is: graphql-transport-ws.`

Comment (bnjjj) [reply docs/source/executing-operations/subscription-support.mdx:164]: I know it's soooo confusing, I don't know what's the right answer here, maybe we could rename that field and use `protocol_name` or I don't know :p I'm open to any suggestions

**Review by lleadbet — APPROVED** [review]

**Review by o0Ignition0o — APPROVED** [review]

**Comment by o0Ignition0o** [comment]
I'm going to fix the conflicts and enable automerge, thanks!

**Comment by o0Ignition0o** [comment]
I'm also adding a changelog entry (happy to edit it before we cut a release!)

**Pull request merged**